internal/core/graphics/image.rs

```diff
@@ -780,4 +788,44 @@ pub(crate) mod ffi {
     ) {
         core::ptr::write(image, Image::from(ImageInner::StaticTextures(textures)));
     }
+
+    #[no_mangle]
+    pub unsafe extern "C" fn slint_image_compare_equal(image1: &Image, image2: &Image) -> bool {
+        return image1.eq(image2);
+    }
+}
+
+/// This structure contains fields to identify and render an OpenGL texture that Slint borrows from the applicatin code.
+/// Use this to embed a native OpenGL texture into a Slint scene, by creating the texture with [`Self::new()`] and
+/// converting it into an [`Image`] using `.into()`.
+///
+/// The ownership of the texture remains with the application. It is the application's responsibility to delete the texture
```

Comment:
Wouldn't it be better to take ownership?
What happens if the application release the texture too early? 
And it seems easy to make a leak.

Reply:
I don't think it's a good idea to offer transfer of ownership at the moment. I mean, for a texture that the application creates itself with glGenTextures that would be possible, but for example when retrieving textures from native APIs, they are only offered by borrow.

For example when mapping a gstreamer video frame to retrieve the GL texture handle, it's only borrowed. Or when retrieving the GL textures from CoreVideo on macOS via CVOpenGLTextureRef, the access to the handle is only borrowed.

In both cases the actual owner is reference counted, so we could try to map this via some type-erased arc that the user has to supply in as well to ensure that the reference count is dropped.

That said, neither Skia's Rust APIs support this at the moment (no mapping for a generic release procedure). Also note how Qt's public API for integrating external textures into QtQuick/RHI has abandoned the ownership concept in their API and leaves it with the application (it's worse there, the old code continues to compile, but the documentation just says that this isn't supported anymore).

So the only other option would be to ask the application to make a copy of the texture, which isn't just a call to some copy function, it requires building an FBO, a rendering pipeline, a vertex shader, a fragment shader, and then running the whole thing.

I'd prefer to make this API explicit about borrowing. If you want I can mark it as `unsafe`, but either way it's low-level. I'm against imposing the requirement to make a copy of the texture to the user.

To answer your question about what happens if the application releases the texture too early: Some weird graphical effects might happen (or often nothing), with OpenGL no crashes though.

We'll run into the same pattern of borrowing textures with Vulkan or Metal, and there a deletion that happens too early will result in crashes.

Reply:
Ok.

> If you want I can mark it as unsafe, but either way it's low-level. I'm against imposing the requirement to make a copy of the texture to the user.

If the use of an invalid texture may result to undefined behavior, it definitively must be `unsafe`.  Otherwise if there is a failure mode (not showing the texture / warning in the console),  then it doesn't need to be unsafe

Reply:
If you pass in garbage, then you'll get undefined behavior. It's not undefined in the main memory sense, but more in whatever the driver does. I don't mind starting with unsafe - at the very least this communicates that it's low-level API.